File: productengineering/persistence/src/main/java/br/com/todeschini/persistence/entities/enums/TipoFilho.java
package br.com.todeschini.persistence.entities.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum TipoFilho implements MappableEnum {

    MDP(1, "MDP"),
    MDF(2, "MDF"),
    ALUMINIO(3, "Alumínio"),
    FUNDO(4, "Fundo");

    private final Integer value;
    private final String label;

    public static TipoFilho fromValue(Integer value) {
        for (TipoFilho obj : TipoFilho.values()) {
            if (obj.value.equals(value)) {
                return obj;
            }
        }
        throw new IllegalArgumentException("Valor desconhecido: " + value);
    }

    @Override
    public int getValue() {
        return value;
    }
}

File: frontend/src/models/enums/tipoFilho.ts
export const DTipoFilhoEnum = {
    MDP: { name: "MDP", value: 1, label: "MDP" },
    MDF: { name: "MDF", value: 2, label: "MDF" },
    ALUMINIO: { name: "ALUMINIO", value: 3, label: "Alumínio" },
    FUNDO: { name: "FUNDO", value: 4, label: "Fundo" },
} as const;

export const getLabel = (name: string): string | undefined => {
    return Object.values(DTipoFilhoEnum).find(item => item.name === name)?.label;
};